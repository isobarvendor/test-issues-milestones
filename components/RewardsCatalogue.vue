<template>
  <div class="rewards-catalogue">
    <div class="wallet-desktop" >
      <div class="wallet-swiper" v-for="(item, index) in data.prizeList" :key="'reward'+index"  @click="goDetail(index)">
        <div class="wallet-swiper-item">
          <img :src="item.imgUrl"
          />
        </div>
        <div class="description bg">
            <div class="name">{{item.name}}</div>
            <div class="expiry" :key="'cost'+index">
              <span>{{item.amountAvailable}}</span>
              <span>Coins</span>
            </div>
          </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: "RewardsCatalogue",
  components: {
  },
  data() {
    return {
      currentPage: 1
    };
  },
  props: {
    data: null,
  },
  computed:{
  	toBeShown() {
    	return this.data.exclusivePrizes.slice(0, this.currentPage * 3);
    },
    totalPages() {
    	return Math.ceil( this.data.exclusivePrizes.length / 3);
    }
  },
  methods:{
    nextPage(){
    	if(this.currentPage <  this.totalPages) this.currentPage++;
    },
    prevPage(){
    	this.currentPage = this.currentPage - 1 || 1;
    },
    goDetail(rewardID){
      this.$router.push({ path: 'rewardsDetail/'+rewardID})
    }
  },
  beforeMount() {},
  mounted(){

  }
};
</script>

<style lang="scss">
.rewards-catalogue{
    margin-bottom: 40px;
    .wallet-desktop{
      display: flex;
      flex-wrap: wrap;
      .wallet-swiper{
        margin: 30px 20px;
        width: 29%;
      }
      .description{
        padding: 20px;
        border-radius: 0px 0px 20px 20px;

        .name:after{
          content: '';
          display: block;
          margin: .5em 0;
          height: 1px;
          background: #585858;

        }
        .expiry{
          display: flex;
          justify-content: flex-end;
          span{
            font-size:12px;
          }
          span:first-child{
            margin-right: 0.5em;
            font-weight: bold;
            font-size: 26px;
          }
        }
      }
    }
    .prize-swiper-pagination > * + *:before{
      background: #585858
    }
    .prize-swiper-pagination .prize-swiper-pagination-bullet{
      background: #585858
    }
    .prize-swiper-pagination .prize-swiper-pagination-bullet-active{
      background: #ffffff
    }
    @media only screen and (max-width: 1199px) {
    .wallet-desktop{
      .wallet-swiper{
          width: 43%;
          margin: 30px 10px;
        .description{
          padding: 1em 10px;
          .name{
            font-size: 14px;
          }
          .expiry{
            justify-content: flex-start;
            span:first-child{
              font-size: 14px;
            }
          }
        }

      }
    }
    }

}
</style>
