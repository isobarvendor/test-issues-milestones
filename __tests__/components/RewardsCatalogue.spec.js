
import { shallowMount,createLocalVue } from '@vue/test-utils'

import RewardsCatalogue from "@/components/RewardsCatalogue";
import Vuex from 'vuex';


let localVue, store;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    store = new Vuex.Store({
      state:{
        language: "id",
        listWallet:{
          walletStatus:[
            {amount :30}
          ]
        }
      }
    });
  });

const factory = () => {
  return shallowMount(RewardsCatalogue, {

    propsData: {
      data :{
        prizeList:[{
          name:"Reward1",
          prizeId:"prizeid",
          prizeCost:[{ammount :100}],
        },
        {
          name:"Reward2",
          prizeId:"prizeid",
          prizeCost:[{ammount :100}],
        }
      ]

      }
    },
    localVue, store
  });
};

describe("RewardsCatalogue", () => {


  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
