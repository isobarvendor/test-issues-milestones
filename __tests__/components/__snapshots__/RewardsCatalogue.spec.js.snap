// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RewardsCatalogue renders properly 1`] = `
<div class="rewards-catalogue">
  <div class="wallet-desktop">
    <div class="wallet-swiper">
      <div class="wallet-swiper-item"><img></div>
      <div class="description bg">
        <div class="name">Reward1</div>
        <div class="">
          <div class="expiry"><span></span> <span>Coins</span></div>
          <div class="btn-area">
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="wallet-swiper">
      <div class="wallet-swiper-item"><img></div>
      <div class="description bg">
        <div class="name">Reward2</div>
        <div class="">
          <div class="expiry"><span></span> <span>Coins</span></div>
          <div class="btn-area">
            <!---->
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
